examples/api/lib/material/carousel/carousel.0.dart

```diff
@@ -35,17 +42,130 @@ class CarouselExample extends StatefulWidget {
 class _CarouselExampleState extends State<CarouselExample> with SingleTickerProviderStateMixin {
   @override
   Widget build(BuildContext context) {
-    return Center(
-      child: ConstrainedBox(
-        constraints: const BoxConstraints(maxHeight: 200),
-        child: Carousel(
-          itemExtent: 330,
-          shrinkExtent: 200,
-          children: List<Widget>.generate(20, (int index){
-            return UncontainedLayoutCard(index: index, label: 'Item $index');
-          }),
+    final double height = MediaQuery.sizeOf(context).height;
+
+    return ListView(
+      children: <Widget>[
+        ConstrainedBox(
+          constraints: BoxConstraints(maxHeight: height / 2),
+          child: Carousel.weighted(
+            controller: CarouselController(initialItem: 1),
```

Comment:
Can you make the controller a class field and dispose it?

Otherwise leak tracking is going to raise its eyebrows :)